# 1. nedēļa: Ievads Machine Learning

In [11]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn

In [12]:
df = sns.load_dataset('titanic')
print(f'Rindu skaits: {df.shape[0]}')
print(f'Kolonnu skaits: {df.shape[1]}')
df.head()

Rindu skaits: 891
Kolonnu skaits: 15


,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [13]:
df.shape

(891, 15)

In [14]:
df.columns.tolist()

['survived',
 'pclass',
 'sex',
 'age',
 'sibsp',
 'parch',
 'fare',
 'embarked',
 'class',
 'who',
 'adult_male',
 'deck',
 'embark_town',
 'alive',
 'alone']

In [15]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    str     
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    str     
 8   class        891 non-null    category
 9   who          891 non-null    str     
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    str     
 13  alive        891 non-null    str     
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), str(5)
memory usage: 80.7 KB


In [16]:
df.describe()

,survived,pclass,age,sibsp,parch,fare
count,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


## A. Problēmas apraksts

Gribam prognozēt, vai Titānika pasažieris izdzīvoja, balstoties uz viņa datiem. Tas palīdzētu saprast, kādi faktori ietekmēja izdzīvošanu. Noderētu kuģniecības kompāniju drošības dienestiem, lai uzlabotu evakuācijas procedūras.

## B. ML uzdevuma tips

Supervised learning, jo datos ir zināmā atbilde - kolonna survived.

Klasifikācija, jo prognozējam 0 vai 1.

## C. Target variable

In [17]:
print(df['survived'].value_counts())
print(f'\nIzdzīvoja: {df["survived"].mean():.1%}')

survived
0    549
1    342
Name: count, dtype: int64

Izdzīvoja: 38.4%


Target mainīgais ir survived. 0 = neizdzīvoja, 1 = izdzīvoja. Aptuveni 38% pasažieru izdzīvoja.

## D. Features saraksts

In [18]:
print('Skaitliskās features:')
print(df.select_dtypes(include='number').columns.tolist())

print('\nKategoriskās features:')
print(df.select_dtypes(include='category').columns.tolist())

Skaitliskās features:
['survived', 'pclass', 'age', 'sibsp', 'parch', 'fare']

Kategoriskās features:
['class', 'deck']


Skaitliskās: age, fare, sibsp, parch

Kategoriskās: sex, pclass, embarked, class, alone

## E. Sākotnējie novērojumi

In [19]:
missing = df.isnull().sum()
missing_pct = (missing / len(df) * 100).round(1)
print('Trūkstošās vērtības:')
print(pd.DataFrame({'Skaits': missing, '%': missing_pct})[missing > 0])

Trūkstošās vērtības:
             Skaits     %
age             177  19.9
embarked          2   0.2
deck            688  77.2
embark_town       2   0.2


In [20]:
print('Izdzīvošana pēc dzimuma:')
print(df.groupby('sex')['survived'].mean().apply(lambda x: f'{x:.1%}'))

Izdzīvošana pēc dzimuma:
sex
female    74.2%
male      18.9%
Name: survived, dtype: str


Kolonnā age trūkst ~20% vērtību, deck kolonnā trūkst ~77% — to varētu izslēgt no modeļa.

Sievietes izdzīvoja daudz biežāk nekā vīrieši (~74% pret ~19%), tāpēc sex būs svarīga feature.